assets/js/modules/adsense/components/module/ModuleTopEarningPagesWidget/index.js

```diff
@@ -70,15 +71,46 @@ function ModuleTopEarningPagesWidget( {
 				limit: 10,
 			};
 
+			const report = select( MODULES_ANALYTICS ).getReport( reportArgs );
+
+			let pageTitles;
+
+			let hasLoadedPageTitles = false;
+			if ( undefined !== report ) {
+				const pagePaths = [];
+				( report?.[ 0 ]?.data?.rows || [] ).forEach(
+					( { dimensions } ) => {
+						if (
+							dimensions.length &&
+							! pagePaths.includes( dimensions[ 0 ] )
```

Comment:
Related to my other comment: The API response includes information about the column headers, i.e. under which `dimensions` index the `ga:pagePath` is stored, so we don't need to have all of this code hard-coded / repeated per widget component.